oracle/datadog_checks/oracle/oracle.py

```diff
@@ -165,44 +142,39 @@ def check(self, _):
 
     @property
     def _connection(self):
-        """Creates a connection or raises an exception"""
         if self._cached_connection is None:
-            if self.can_use_oracle_client():
-                self._cached_connection = self._oracle_client_connect()
-            elif JDBC_IMPORT_ERROR:
-                self._connection_errors += 1
+            if self.can_use_jdbc():
+                try:
+                    self._cached_connection = self._jdbc_connect()
+                except Exception as e:
+                    self.log.error("The JDBC connection failed with the following error: %s", str(e))
+                    self._connection_errors += 1
+            else:
+                self._cached_connection = self._oracle_connect()
+        return self._cached_connection
+
+    def can_use_jdbc(self):
+        if self._jdbc_driver:
+            if JDBC_IMPORT_ERROR is not None:
                 self.log.error(
-                    "Oracle client is unavailable and the integration is unable to import JDBC libraries. You may not "
+                    "The integration is unable to import JDBC libraries. You might not "
                     "have the Microsoft Visual C++ Runtime 2015 installed on your system. Please double check your "
                     "installation and refer to the Datadog documentation for more information."
```

Comment:
```suggestion
                    "The integration is unable to import JDBC libraries. Please double check your "
                    "installation and refer to the Datadog documentation for more information."
```
careful here, visual c++ was required for the instant client, not jdbc. In fact jdbc is linux only

Reply:
Ah I was just trying to reuse the error that we generated for the JDBC import error but that's good to know